### ISSUE TITLE
Timetabler ersetzen

### ISSUE THREAD

**Issue by coderbyheart** [BCRM/www#7]
# Mockups
## Session Plan (Tablet)

![sessionplantablet](https://cloud.githubusercontent.com/assets/188915/8893701/23e96ffe-339c-11e5-89ae-7142c1244f93.png)
## Session Plan (Smartphone)

![sessionplansmartphone](https://cloud.githubusercontent.com/assets/188915/8893703/2bb985b6-339c-11e5-9aba-ee111cb8c729.png)
## Session Plan (Detailansicht Smartphone)

![sessionplansmartphonedetail](https://cloud.githubusercontent.com/assets/188915/8893704/2edf6850-339c-11e5-8c80-b06c53f48563.png)
## Eingabeformular (Website)

![02_session_form](https://cloud.githubusercontent.com/assets/188915/8893723/eda85e36-339c-11e5-9c50-ca95caf3b5d9.png)

![03_session_sent](https://cloud.githubusercontent.com/assets/188915/8893724/f012797c-339c-11e5-82bb-dedce2d7d9fc.png)

![04_sessionradar_session](https://cloud.githubusercontent.com/assets/188915/8893725/f25a6c08-339c-11e5-9fb4-92745ceabdb3.png)

**Comment by coderbyheart** [comment]
@ravetracer Der @martinkraft kümmert sich um die Ausgabe des Session-Plans, d.h. am Wochende werden dann die Session-Daten mit den Räumen und Zeiten verknüpft um diese Ausgabe zu erstellen. Du kannst also aus den Screens ablesen, welche Infos wir von den Usern brauchen. 

Wichtige Zusatzinfos für den Timetabler wären noch Hilfreich (siehe BCRM/sessionOpt#1):

- Raumgröße (mind. Anzahl Personen für diese Session)
- Beamer (ja / nein)

**Comment by coderbyheart** [comment]
@McWizard @martinkraft Oben findet ihr auch die Screens von @ravetracer, er kümmert sich aktuell darum die Session-Infos zu erfassen.

**Comment by intocommerce** [comment]
Sehr cool!

Wir müssen nur bedenken, dass wir eine ordentliche Druckansicht bekommen, damit die Sessions einzeln und später auch der Plan ausgedruckt werden kann.

Viele Grüße, 
Matthias

**Comment by Ravetracer** [comment]
@coderbyheart Heißt das, dass die Sessionvorschläge, welche den Ersatz für das Piratenpad darstellen, dann direkt in den neuen(?) Timetabler übergehen werden?

**Comment by coderbyheart** [comment]
Ja, das wäre ideal. Letztendlich braucht es dafür noch ein Admin-Frontend in dem Session-Vorschläge den Timeslots zugeordnet werden.

**Comment by Ravetracer** [comment]
Puh, wenig Zeit zur Zeit ;-).
Was nun geht ist:
- Sessionvorschlag eintragen und ansehen
- Sessionvorschlag ändern (falls man etwas vergessen hat etc.)
- Mail an den Ansprechpartner mit Bearbeitungslink (hat ein wenig gedauert, da ich Swiftmail irgendwie falsch konfiguriert hatte).

Fragen:
- Was soll ich denn testtechnisch machen? Ich habe bisher noch nie Tests geschrieben und bekomme da auch nicht so richtig den Anfang.

- Soll es auch möglich sein, eine Session wieder zu entfernen oder nur zu canceln (heißt, sie wird im Radar angezeigt, aber als "abgesagt", so dass die Besucher immer noch sehen können, was vorgeschlagen wurde)

- Laut den Screenshots kann man mehrere Ansprechpartner definieren. Habe jetzt nur einen drin, der in der Mail auch angesprochen wird. Soll das noch angepasst werden, dass man mehrere eintragen kann?

- Die Mail ist aktuell nur so'n bisschen Text mit Link. Gibt es irgendwo eine Headergrafik, die man zum Aufhübschen einbauen könnte?

**Comment by coderbyheart** [comment]
> Was soll ich denn testtechnisch machen? Ich habe bisher noch nie Tests geschrieben und bekomme da auch nicht so richtig den Anfang.

Wenn Testen Neuland für dich ist, dann lass es erstmal weg. Dieser Teil der Anwendung ist ja nicht so kritisch.

> Soll es auch möglich sein, eine Session wieder zu entfernen oder nur zu canceln (heißt, sie wird im Radar angezeigt, aber als "abgesagt", so dass die Besucher immer noch sehen können, was vorgeschlagen wurde)

Session sollten löschbar sein (durch den Owner). Aber den Status "abgesagt" bilden wir erstmal nicht ab.

> Laut den Screenshots kann man mehrere Ansprechpartner definieren. Habe jetzt nur einen drin, der in der Mail auch angesprochen wird. Soll das noch angepasst werden, dass man mehrere eintragen kann?

Ist doch ein Freitext-Feld, oder? Da kann man ja mehrere Namen / Twitter-Handles eintragen.

> Die Mail ist aktuell nur so'n bisschen Text mit Link. Gibt es irgendwo eine Headergrafik, die man zum Aufhübschen einbauen könnte?

Nein, wir versenden nur Plain-Text-Mails.

**Comment by coderbyheart** [comment]
An alle: Heute Abend um 2030 ist der nächste Hangout zu dem Thema, wer kann nicht?

**Comment by martinkraft** [comment]
Ich glaube es wäre besser die Daten der Ansprechpartner in einzelnen Feldern zu haben. Zum einen, weil das ja wichtig ist um bei der Sessionverteilung Kollisionen zu vermeiden (also mehrere Sessions der selben Person zur selben Zeit) und zum anderen weil wir nur so die Mail und Twitter-IDs verifizieren und ordentlich verlinken können.

**Comment by Ravetracer** [comment]
Bin bereit für den Hangout, da können wir das ja noch mal klären.

**Comment by martinkraft** [comment]
Helge und ich sind gerade im Hangout. Markus scheint's verpennt zu haben und wir haben leider Deine Kontaktdaten nicht.
Wie ist denn Deine Google-ID?

**Comment by Ravetracer** [comment]
Kleine Zusammenfassung vom Hangout gestern:
- es können 1-n Ansprechpartner definiert werden, jeweils mit Name, Mail, Twitterhandle
- Session proposals bekommen eine ID (in welcher Form auch immer), welche es vor Ort möglich macht, schnell darauf zuzugreifen, falls es Umstrukturierungen etc. gibt
- zusätzliche Informationen wie Wunschdatum (inkl. Uhrzeit), Beamer etc. sollen erfasst werden
- Ausgabeformat für das autom. Sessionverteilungsscript muss erstellt werden

Wie weit ich anschließend den Timetabler noch anpacken kann, weiß ich bisher nicht, da meine Zeit leider immer sehr knapp ist. Aber schauen wir mal.
